authentication.mongoose.js

```diff
@@ -0,0 +1,71 @@
+/*
+@author: Emad Bin Abid
+@date: July 03, 2018
+*/
+
+//Dependencies
+const config = require('./config');
+const userModel = require('./models/user-model/user.model');
+
+const UserModel = userModel.UserModel;
+
+//Validating User on /login request.
+/*
+method: validateUser(expressInstance, jwtInstance, userRouterInstance)
+url: domain/login
+request object: expects a json object of type { "user": object }
+response object: sends a json object of type { "user": object, "token": token }. If error, then sends "Unauthorized"
+*/
+exports.validateUser = function(expressInstance, jwtInstance)
+{
+    //Validating User
+    expressInstance.post('/login', (req, res) => 
+    {
+        UserModel.findOne(req.body.user, (err, dbObject) => 
+        {
+            if(err)
+            {
+                res.status(401).send('Unauthorized');
+            }
+            else
+            {
+                if(dbObject === null)
```

Comment:
rename `dbObject` to `user` maybe. better to keep variable names such that they reflect what they are